vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/cast/AbstractCastToOperation.java

```diff
@@ -226,4 +226,8 @@ public void convertNormalizedString(UTF8StringPointable stringp, DataOutput dOut
     public void convertToken(UTF8StringPointable stringp, DataOutput dOut) throws SystemException, IOException {
         throw new SystemException(ErrorCode.XPTY0004);
     }
+
+    public void convertNull(DataOutput dOut) throws SystemException, IOException {
```

Comment:
Also, add a direct test for casting:
xs:string("test") cast as js:null

Change the query as necessary for your test.